File: 새 폴더/Battle.js
function getRandomAttackValue(attack) {
    var random = Math.floor(Math.random() * attack) + 1;
    return random;
}

function battleOneByOne (target){
    var monsterDamage = getRandomAttackValue(target.attack);
    Jimmy.hp = Jimmy.hp - monsterDamage;
        tvSystem(target.name + "가" + Jimmy.name + "에게 데미지를 " + monsterDamage + "만큼 입혔습니다.");
    var playerDamage = getRandomAttackValue(Jimmy.attack);
    target.hp = target.hp - playerDamage;
    br();
    tvSystem(Jimmy.name + "가" + target.name + "에게 데미지를 " + playerDamage + "만큼 입혔습니다.");
    displayRoomMonstersInfo();
    displayCharacterInfo();
    hr();
    turn();

    if(target.hp<=0){
        getReward(target);
        monsterDeleteFromMonsterArray(target);
        displayRoomMonstersInfo();
        return;
    }
}

function turnBattle() {
        var hostileMonsters = getRoomMonstersHostile();
        var target = hostileMonsters[0];
        battleOneByOne(target);

    if(Jimmy.hp <=0){
        tvSystem("죽었습니다.");
        br();
        tvSystem("[GAME OVER]");
        hr();
        tvObject("[GAME OVER]");
        tvPlayer("[GAME OVER]");
        mode = "사망";
    }

    hostileMonsters = getRoomMonstersHostile();
    if(hostileMonsters.length==0){
        battleEnd();
        return;
    }
    return;
}

function battleEnd() {
    tvSystem("전투 종료");
    hr();
    mode = "대기";
}

function monsterDeleteFromMonsterArray(dieMonster) {
    var findI = -1;
    for(var i=0;i<monsterArray.length;i++){
        if(monsterArray[i].id == dieMonster.id){
            findI = i;
        }
    }
    if(findI != -1){
        monsterArray.splice(findI,1);
    }
}

function getReward(dieMonster){
    Jimmy.exp = Jimmy.exp + dieMonster.exp;
    displayPlayerExp();
    levelUp();

    tvSystem(dieMonster.name + "를 처치했습니다. 경험치를 " + dieMonster.exp + "만큼 얻었습니다.");
    hr();
    var randomCoin = Math.floor(Math.random()*dieMonster.coin) + 1;
    Jimmy.coin = Jimmy.coin + randomCoin;
    tvSystem("코인을 " + randomCoin + "만큼 얻었습니다.");
    hr();
    return;
}

function objectClick(objectId){
    if(mode == "전투") {
        console.log("이미 전투 상태입니다.");
        return;
    }

    console.log("클릭한 오브젝트 ID:"+ objectId);
    mode = "전투";

    for(var i=0;i<monsterArray.length; i++){
        if(monsterArray[i].id == objectId){
            monsterArray[i].aggressionType = "H";
        }
    }
    displayRoomMonstersInfo();
    turn();
}

File: JavaRpg_V0.2/src/com/cre/Battle.java
package com.cre;

import com.cre.util.Util;

public class Battle {
	static Monster[] monsterArr;
	static Monster target;

	static Monster[] getMonsterArray(Monster[] monsterArray) {
		monsterArr = monsterArray;
		return monsterArr;
	}

	static boolean game = true;

	static void battleOneByOne(int currentRoomId) {
		target = RoomFunctions.roomMonsters.get(0);

		int monsterDamage = Util.random(target.attack);
		int characterDamage = Util.random(Character.attack);

		Character.hp = Character.hp - monsterDamage;
		Util.prl("<시스템> " + target.name + "이(가) " + Character.name + "에게 " + monsterDamage + "만큼 데미지를 입혔습니다.");
		target.hp = target.hp - characterDamage;
		Util.prl("<시스템> " + Character.name + "이(가) " + target.name + "에게 " + characterDamage + "만큼 데미지를 입혔습니다.");
		Util.br();
		if (Character.hp <= 0) {
			battleEnd(target);
			return;
		} else if (target.hp <= 0) {
			battleEnd(target);
			return;
		}

		Turn.turn++;
	}

	static void battleEnd(Monster target) {
		if (target.hp <= 0) {
			RoomFunctions.roomMonsters.remove(0);
			deleteDieMonster(target);
			Util.prl("<시스템> " + Character.name + "이(가) " + target.name + "을(를) 처치했습니다.");
			battleGetReward(target);
			Character.lvUpCheck();
			Util.br();
		} else if (Character.hp <= 0) {
			Util.prl("<시스템> 죽었습니다.");
			Util.prl("<시스템> GameOver");
			Util.br();
			isGameOver(false);
		}
	}

	static void deleteDieMonster(Monster dieMonster) {
		int findI = -1;
		for (int i = 0; i < monsterArr.length; i++) {
			if (monsterArr[i] != null) {

				if (dieMonster.id == monsterArr[i].id) {
					findI = i;
				}
			}
		}
		if (findI != -1) {
			monsterArr[findI] = null;
		}
	}

	static boolean isGameOver(boolean game) {
		if (game == false) {
			Battle.game = false;
		} else {
			Battle.game = true;
		}
		return Battle.game;
	}

	static void battleGetReward(Monster target) {
		Character.exp += target.exp;
		Util.prl("<시스템> 경험치를 얻었습니다.(" + target.exp + ")");
	}

}
